manifests/init.pp

```diff
@@ -72,10 +73,20 @@
       refreshonly => true,
     }
 
-    create_resources('firewalld_port',      $ports)
-    create_resources('firewalld_zone',      $zones)
-    create_resources('firewalld_service',   $services)
-    create_resources('firewalld_rich_rule', $rich_rules)
+    $defaults = { zone => $default_zone }
+
+    if $zones {
+      create_resources('firewalld_zone', $zones)
+    }
+    if $ports {
```

Comment:
This won't be an issue in 3.0.0 since we're dropping `create_resources` altogether in favor of 4.x parser iterators.